elements/drag-drop-elements/pl-drag-drop-element.py

```diff
@@ -0,0 +1,389 @@
+import prairielearn as pl
+import lxml.html
+import random
+import chevron
+import base64
+import os
+
+# Read https://prairielearn.readthedocs.io/en/latest/devElements/
+# Official documentation on making custom PL
+
+
+def render_html_colour(score):
+    # used to render the correct colour depending on student score
+    if score == 0:
+        return 'badge-danger'
+    elif score == 1.0:
+        return 'badge-success'
+    else:
+        return 'badge-warning'
+
+# courtesy of https://github.com/PrairieLearn/PrairieLearn/pull/2625
+
+
+def getallAnswer(submitted_blocks, block_indents, leading_code, trailing_code):
+    if len(submitted_blocks) == 0:
+        return ''
+    answer = ''
+    for index, answer in enumerate(submitted_blocks):
+        indent = int(block_indents[index])
+        answer += ('    ' * indent) + submitted_blocks[index] + '\n'
+    answer = leading_code + '\n' + answer + trailing_code + '\n'
+    return answer
+
+
+def gettarilAnswer(submitted_blocks, block_indents, trailing_code):
+    if len(submitted_blocks) == 0:
+        return ''
+    answer = ''
+    for index, answer in enumerate(submitted_blocks):
+        indent = int(block_indents[index])
+        answer += ('    ' * indent) + submitted_blocks[index] + '\n'
+    answer = answer + trailing_code + '\n'
+    return answer
+
+
+def getleadAnswer(submitted_blocks, block_indents, leading_code):
+    if len(submitted_blocks) == 0:
+        return ''
+    answer = ''
+    for index, answer_indent in enumerate(block_indents):
+        indent = int(answer_indent)
+        answer += ('    ' * indent) + submitted_blocks[index] + '\n'
+    answer = leading_code + '\n' + answer
+    return answer
```

Comment:
I think it was noted in the original PR for this, that these functions are really similar and so they should be combined or abstracted in some way.

Reply:
I think github auto-resolved this comment due to some other changes, but this issue hasn't been addressed yet.

Reply:
commenting here or else github would bury this review even tho it's not resolved